examples/cg/cg.py

```diff
@@ -0,0 +1,91 @@
+import argparse
+import contextlib
+import time
+
+import numpy as np
+from scipy.sparse.linalg import cg
+import six
+
+import cupy
+
+
+@contextlib.contextmanager
+def timer(message):
+    cupy.cuda.Stream.null.synchronize()
+    start = time.time()
+    yield
+    cupy.cuda.Stream.null.synchronize()
+    end = time.time()
+    print('%s:  %f sec' % (message, end - start))
+
+
+def fit(A, b, x0, tol, max_iter):
+    xp = cupy.get_array_module(A)
+    x = x0
+    r0 = b - xp.dot(A, x)
+    p = r0
+    for i in six.moves.range(max_iter):
+        a = xp.dot(r0.T, r0) / xp.dot(xp.dot(p.T, A), p)
+        x = x + p * a
+        r1 = r0 - xp.dot(A * a, p)
+        if xp.linalg.norm(r1) < tol:
+            return x
+        b = xp.dot(r1.T, r1) / xp.dot(r0.T, r0)
+        p = r1 + b * p
+        r0 = r1
+    msg = 'Failed to converge. Increase max-iter or tol.'
+    print(msg)
+    return x
+
+
+def run(gpuid, tol, max_iter):
+    # Solve simultaneous linear equations, Ax = b.
+    # To check whether the answers are correct, each 'b' is printed.
```

Comment:
"each b" in the sentence seems ambiguous to me. There are two types of `b`: the input of the algorithm (i.e. the true value) and `b` computed from `A` and the solution of CG by matrix multiplication. Which do you think of in mind?